service/src/main/java/bio/terra/landingzone/common/utils/LandingZoneFlightBeanBag.java

```diff
@@ -20,15 +23,18 @@ public class LandingZoneFlightBeanBag {
   private final LandingZoneBillingProfileManagerService bpmService;
   private final ObjectMapper objectMapper;
 
+  @Lazy
   @Autowired
   public LandingZoneFlightBeanBag(
+      LandingZoneService landingZoneService,
```

Comment:
Nit: I wish there was better (any) documentation in this repo and better naming. We are already in "landingzone", so why is everything prefixed with "LandingZone"… and more importantly, what is the difference between LandingZoneService and LandingZoneDao and LandingZoneManagerProvider, etc.

I don't expect you necessarily to answer, but I'm feeling a bit lost as I wade into this code.

Reply:
The starting point here is LandingZoneService. It's like entry point. Basically, it initiates stairway-based flights. LandingZoneDao cares about saving LZ data into db. LandingZoneManagerProvider instantiates LandingZoneManager depending on the request (managed resource group, etc.).